### ISSUE TITLE
URIError: URI malformed at encodeURIComponent  (UNICODE problem)

### ISSUE THREAD

**Issue by hfrmobile** [DavidAnson/markdownlint#2069]
False positive with VS Extension:

<img width="907" height="410" alt="Image" src="https://github.com/user-attachments/assets/f3b1d2d6-d4d9-42c0-bf3d-742d725bb3d7" />

With markdownlint crash:

```text
PS> markdownlint .\English.md              
URIError: URI malformed
    at encodeURIComponent (<anonymous>)
    at Object.MD051 [as function] (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/md051.mjs:123:35)
    at invokeRuleFunction (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:708:53)
    at forRule (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:719:9)
    at Array.map (<anonymous>)
    at lintContentInternal (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:746:46)
    at lintContent (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:765:5)
    at lintContentWrapper (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:815:12)
    at lintFile (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:832:5)
    at lintWorker (file:///C:/Users/User/AppData/Roaming/npm/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.mjs:923:7)
```

Path: ....Markdown\Englisch\English.md

(the First "Englisch" is no typo, it is on purpose -- German name for "Englisch").

Renaming the Markdown file or removing and retyping the text doesn't solve it. File is UTF-8 (Windows CRLF). Having other files with same situation but without that problem (false positive / crash) ...

**Comment by hfrmobile** [comment]
I figured out what is causing the problem:

```markdown
- [Practical language use](<🕮 Practical language use.md>)
```

It's the 🕮 what causes the problem. When I remove the "🕮" the problem doesn't occur... (but then the link doesn't work because the file is named "🕮 Practical language use.md") --> Using the "🕮" UNICODE Symbol to force special sorting (so it will be shown as the very first file in VS Code Explorer, even before "AAA.md"...)

I remember now that I had an issue with some UNICODE-Symbols/Characters in the past...
